### PR TITLE
ARXIVNG-109: isolated advanced search in routes, controllers, templates

### PR DESCRIPTION
**Pull request by erickpeirson**
Highlights:
- Split controllers.py into controllers/simple.py and controllers/advanced
- Moved forms.py to controllers/advanced, since it's really only used there
- Moved advanced-form-specific bits from domain.Query to domain.AdvancedQuery
- Added /advanced route in routes.ui
- Forked templates/search.html to advanced_search.html
- Tweaks and twists to the advanced template, macros

### CODE REVIEW

**Comment by erickpeirson** [comment]
image

**Comment by erickpeirson** [comment]
image

File: search/templates/search/search-macros.html [hunk 1]
```diff
@@ -55,159 +55,40 @@
...
+        </div>
+        <div class="select is-small">
+          <select>
+            <option>All fields</option>
+            <option>Author</option>
+            <option>Date</option>
+          </select>
+        </div>
       </div>
-    </div>
```
Comment (eawoods) [inline search/templates/search/search-macros.html]: It looks like two <div> openings got taken out but only one </div> removed? Hard to see from this view.

Comment (erickpeirson) [reply search/templates/search/search-macros.html]: image

File: search/domain/base.py [hunk 2]
```diff
@@ -79,10 +79,15 @@ def valid(self):
             return False
         return True
 
-    def __str__(self):
+    def json(self):
```
Comment (mhl10) [inline search/domain/base.py]: Looks good. I think ```populate_test_metadata.py``` would need a tiny update:
```
        if print_indexable:
            print(document.json) 
```

Comment (erickpeirson) [reply search/domain/base.py]: Whoops, yes

**Pull request merged**